v2/src/features/UploadList/components/UploadButtonBase.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import Tooltip from '@mui/material/Tooltip';
+import BackupIcon from '@mui/icons-material/Backup';
+import { Button } from '@mui/material';
+
+type Props = {
+  onClick: (fileList: FileList) => void;
+};
+
+export default function UploadButtonBase({ onClick }: Props) {
+  const inputFileRef = React.useRef<HTMLInputElement | null>(null);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      const file = e.target.files;
```

Comment:
Be careful this is an array of files: https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications#accessing_selected_files_on_a_change_event

This is not an issue now since `multiple=false` below but it's still gonna be an array: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#multiple

Reply:
I'd use something similar to current logic to handle the case where multiple files are selected (if we set `multiple=true` later).

We can then either bundle all files into a single file, or loop through all files and handle upload for each one.

Reply:
Im aware that is an array of file and this is being handled at UploadList

Reply:
I think it was a typo in the const name, because if you check the type:
```onClick: (fileList: FileList) => void;``` Im specifically expressing there that it is a fileList